### PR TITLE
makefile(all): add format and check targets for makefiles

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/6493

See more: https://github.com/pingcap/tiflow/pull/6595#discussion_r936421163

### What is changed and how it works?
Add format and check targets for makefiles. 

#### Tests 

 - No code

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by hi-rustin** [comment]
/hold 
We should also use SED in other targets.

**Review by lonng — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -120,6 +130,14 @@ test: unit_test dm_unit_test engine_unit_test
 
 build: cdc dm engine
 
+check-makefiles: ## Check the makefiles format.
+check-makefiles: format-makefiles
+	@git diff --exit-code -- $(MAKE_FILES) || (echo "Please format Makefiles by running 'make format-makefiles'" && false)
+
+format-makefiles: ## Format all Makefiles.
+format-makefiles: $(MAKE_FILES)
+	$(SED) -i -e 's/^\(\t*\)  /\1\t/g' -e 's/^\(\t*\) /\1/' -- $?
```
Comment (amyangfei) [inline Makefile:139]: No need to address. It works fine if the indent of tab is 2 or 4 spaces, but in 4 spaces scenario we need to run `make format-makefiles` multiple times to replace all spaces.

**Review by amyangfei — APPROVED** [review]

**Comment by amyangfei** [comment]
> /hold We should use SED in other targets.

What does other targets mean

**Comment by hi-rustin** [comment]
> > /hold We should use SED in other targets.
> 
> What does other targets mean

I meant the target which uses the sed. Like `dm_coverage`.

**Comment by hi-rustin** [comment]
/unhold

Ready to merge.

File: Makefile (hunk 1 above)
Comment (hi-rustin) [reply Makefile:139]: Good catch! 
Thanks for your review!  💚 💙 💜 💛 ❤️

**Pull request merged**